### PR TITLE
Upgrade Python and Pytorch versions for some examples

### LINKED ISSUE #1883: Upgrade Python and Pytorch version in container image for pytorch-mnist example
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
As discussed in https://github.com/kubeflow/katib/pull/1882#discussion_r889196264, we need to upgrade Python and Pytorch version.

/cc @johnugeorge 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

---

Love this feature? Give it a 👍 We prioritize the features with the most 👍

### PR DESCRIPTION
**Pull request by tenzen-y**
**What this PR does / why we need it**:
I upgraded Python and Pytorch versions for `darts-cnn-cifar10` and `pytorch-mnist` examples.

**Which issue(s) this PR fixes** _(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)_:
Fixes #1883 

**Checklist:**

- [ ] [Docs](https://www.kubeflow.org/docs/components/katib/) included if any changes are user facing

### CODE REVIEW

**Comment by tenzen-y** [comment]
@johnugeorge 
This improvement would be better to include in the next Katib release.

File: examples/v1beta1/trial-images/darts-cnn-cifar10/architect.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+import numpy as np
```
Comment (anencore94) [inline examples/v1beta1/trial-images/darts-cnn-cifar10/architect.py:14]: Do we need this line ?

**Review by anencore94 — APPROVED** [review]
/lgtm
It built well in amd64, cpu, linux

File: examples/v1beta1/trial-images/darts-cnn-cifar10/architect.py (hunk 1 above)
Comment (tenzen-y) [reply examples/v1beta1/trial-images/darts-cnn-cifar10/architect.py:14]: Nice catch!
Thanks, @anencore94!

Comment (tenzen-y) [reply examples/v1beta1/trial-images/darts-cnn-cifar10/architect.py]: Done.

Comment (anencore94) [reply examples/v1beta1/trial-images/darts-cnn-cifar10/architect.py]: Thanks :)

**Review by anencore94 — APPROVED** [review]
/lgtm

**Comment by djmcgreal-cc** [comment]
Since the base image is no longer built with GPU support, CUDA acceleration is not available. Since we were using this image in our tests, they broke. It's misleading because the mnist.py still tries to use cuda but it will never be available.

**Comment by johnugeorge** [comment]
@tenzen-y cuda image resources are not available with Github actions?

**Comment by tenzen-y** [comment]
> @tenzen-y cuda image resources are not available with Github actions?

@johnugeorge Since the image with CUDA support size is very heavy as shown below, we will face `no space left on device: unknown` error.

Not CUDA support image size: [639.63 MB](https://hub.docker.com/layers/kubeflowkatib/pytorch-mnist/v1beta1-72fff88/images/sha256-f40e2d2f636b00c130cef048714fe616852070eeb1bf7bcd84ea1dcd010b5529?context=explore)
CUDA support image size: [1.2 GB](https://hub.docker.com/layers/kubeflowkatib/pytorch-mnist/v1beta1-5c7dce6/images/sha256-f4c5bc9b78bba2fd85c4eee0477df153a45cfac84b6fe41abe057384b2b39662?context=explore)

**Comment by johnugeorge** [comment]
@tenzen-y We can then defer this building step till we find a solution. eg: CI in AWS

Other option, is to manually build and push image as changes are not frequent to this build

**Comment by tenzen-y** [comment]
@johnugeorge 
We can build the image in [this](https://github.com/kubeflow/katib/blob/8f182c2373e39292e316ade07779ea5a84c48b28/.github/workflows/publish-algorithm-images.yaml) workflow, although we can not run an operation check of the image in [this](https://github.com/kubeflow/katib/blob/8f182c2373e39292e316ade07779ea5a84c48b28/.github/workflows/pytorch-mnist-e2e-test.yaml) workflow.

So, we cannot confirm that the image works, but we can provide the image with support CUDA built in gh-actions.

**Comment by johnugeorge** [comment]
@tenzen-y I feel, that should be ok for now

**Comment by tenzen-y** [comment]
@johnugeorge 
I see.
I will create a PR to add the `pytorch-mnist-gpu` image by the end of this week.

**Pull request merged**